### PR TITLE
Doc draft PRs for WIP, private forks for CI-bashing

### LINKED ISSUE #157: Document WIP/Draft PR practices
Per discussion with @mkolesnik et al, we should add something like the following to our Code Review docs.

- GitHub has support for draft PRs (like we know from Gerrit). Use these for WIP patches.
- For PRs that should never be merged, ie for testing, you can push to your fork. This will keep the PR record clean and count towards your GHA jobs, not the main SubM jobs.

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

**Review by nyechiel — APPROVED** [review]

File: src/content/development/code-review/_index.en.md [hunk 1]
```diff
@@ -89,6 +89,20 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
 other relevant code has changed, the new job results don't tell us anything new. Since changes are constantly being merged, it's likely
 another merge commit and set of jobs will be necessary right before merging anyway.
 
+### Mark work-in-progress PRs as drafts
+
+To clearly indicate a Pull Request is still under development, not yet ready for review, mark it as a draft. It is not necessary to modify
```
Comment (tpantelis) [inline src/content/development/code-review/_index.en.md]: We reference _PR_ before this so I think we can use it here too.

```suggestion
To clearly indicate a PR is still under development and not yet ready for review, mark it as a draft. It is not necessary to modify
```

File: src/content/development/code-review/_index.en.md [hunk 2]
```diff
@@ -89,6 +89,20 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
...
+PR summaries or commit messages ("WIP" or "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the
```
Comment (tpantelis) [inline src/content/development/code-review/_index.en.md]:
```suggestion
PR summaries or commit messages to indicate this (eg "WIP" or "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the
```

File: src/content/development/code-review/_index.en.md [hunk 3]
```diff
@@ -89,6 +89,20 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
...
+commit message you plan to merge will allow gitlint to verify it. PRs should typically be marked as drafts if any of the required CI is
+failing.
```
Comment (tpantelis) [inline src/content/development/code-review/_index.en.md]: I'm not clear on the last sentence. If I read it correctly, if a CI job fails that prevents merging we should change it to a draft even if it's ready for review?

File: src/content/development/code-review/_index.en.md [hunk 4]
```diff
@@ -89,6 +89,20 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
...
+
+### Use private forks for debugging PRs by running CI
+
+If a PR is not expected to pass CI but the author wants to see the results to enable development, use a personal fork to run CI. This avoids
+clogging the GitHub Actions job queue of the Submariner-io GitHub Organization. After the same `git push` to your personal fork you'd
+typically do for a PR, simply choose your fork as the "base repository" of the PR in GitHub's "Open a pull request" UI. After creating the
+PR, CI will trigger as usual but the jobs will count towards your personal queue.
+
```
Comment (tpantelis) [inline src/content/development/code-review/_index.en.md:109]: Cool - I didn't know this trick!

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/development/code-review/_index.en.md (hunk 2 above)
Comment (dfarrell07) [reply src/content/development/code-review/_index.en.md]: I had a version like this, but it seemed unnecessarily verbose. Also the "this" is hanging on to a broad idea from the first sentence, which is more difficult to read than necessary IMHO.

File: src/content/development/code-review/_index.en.md (hunk 3 above)
Comment (dfarrell07) [reply src/content/development/code-review/_index.en.md]: I'll try to reword it to be more clear.

File: src/content/development/code-review/_index.en.md (hunk 4 above)
Comment (dfarrell07) [reply src/content/development/code-review/_index.en.md:109]: Added another line to further clarify.

File: src/content/development/code-review/_index.en.md (hunk 2 above)
Comment (dfarrell07) [reply src/content/development/code-review/_index.en.md]: I did part of this in the latest.

**Review by tpantelis — APPROVED** [review]

**Comment by tpantelis** [comment]
I'll give others a chance to review.

File: src/content/development/code-review/_index.en.md [hunk 5]
```diff
@@ -89,6 +89,21 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
 other relevant code has changed, the new job results don't tell us anything new. Since changes are constantly being merged, it's likely
 another merge commit and set of jobs will be necessary right before merging anyway.
 
+### Mark work-in-progress PRs as drafts
+
+To clearly indicate a PR is still under development and not yet ready for review, mark it as a draft. It is not necessary to modify PR
+summaries or commit messages (e.g. "WIP", "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the
+commit message you plan to merge will allow gitlint to verify it. PRs should typically be marked as drafts if any CI is failing that the
+author can fix before asking for code review.
+
```
Comment (skitt) [inline src/content/development/code-review/_index.en.md:98]:
```suggestion
Please do this when opening the PR: instead of clicking on the “Create pull request” button, click on the drop-down arrow next to it, and select “Create draft pull request”. This will avoid notifying code owners; they will be notified when the PR is marked as ready for review.
```

File: src/content/development/code-review/_index.en.md [hunk 6]
```diff
@@ -89,6 +89,21 @@ ready to merge (has [required reviews](#two-non-author-committer-approvals-requi
...
+### Use private forks for debugging PRs by running CI
+
+If a PR is not expected to pass CI but the author wants to see the results to enable development, use a personal fork to run CI. This avoids
+clogging the GitHub Actions job queue of the Submariner-io GitHub Organization. After the same `git push` to your personal fork you'd
+typically do for a PR, simply choose your fork as the "base repository" of the PR in GitHub's "Open a pull request" UI. After creating the
+PR, CI will trigger as usual but the jobs will count towards your personal queue. You will need to open a new PR against the main repository
+once your proposed change is ready for review.
+
```
Comment (skitt) [inline src/content/development/code-review/_index.en.md:109]:
```suggestion
Remember to keep your fork’s main branch up-to-date; this will make your local PR easier to understand.
```

**Review by skitt — CHANGES_REQUESTED** [review]
A couple of suggestions.

File: src/content/development/code-review/_index.en.md (hunk 5 above)
Comment (dfarrell07) [reply src/content/development/code-review/_index.en.md:98]: Ah nice, I didn't know this!

**Review by skitt — APPROVED** [review]
